src/app/beer_garden/api/stomp/manager.py

```diff
@@ -1,68 +1,98 @@
-from beer_garden.events.processors import QueueListener
-from beer_garden.api.stomp.server import Connection
-from brewtils.stoppable_thread import StoppableThread
+from beer_garden.events.processors import BaseProcessor
+
+logger = logging.getLogger(__name__)
+
 
+class StompManager(BaseProcessor):
+    """Manages Stomp connections and events for the Stomp entry point
 
-class StompManager(StoppableThread):
-    logger = logging.getLogger(__name__)
+    Will poll the multiprocessing.Connection for incoming events, and will invoke the
+    handle_event method for any received.
+
+    Also functions as the entry point's Event Manager. It simply sends any generated
+    events across the multiprocessing.Connection.
+
+    """
 
     @staticmethod
-    def connect(stomp_config, gardens):
+    def connect(stomp_config: Box) -> Connection:
+        """Create and return a stomp connection """
+
         conn = Connection(
-            host_and_ports=[(stomp_config.get("host"), stomp_config.get("port"))],
+            host=stomp_config.get("host"),
+            port=stomp_config.get("port"),
             send_destination=stomp_config.get("send_destination"),
             subscribe_destination=stomp_config.get("subscribe_destination"),
             ssl=stomp_config.get("ssl"),
             username=stomp_config.get("username"),
             password=stomp_config.get("password"),
         )
-        conn.connect(connected_message="connected", gardens=gardens)
+
+        if conn.connect():
+            logger.info("Successfully connected")
+        else:
+            logger.info("Failed to connect")
+
         return conn
 
-    def __init__(self, ep_conn=None, stomp_config=None, name=None, is_main=True):
-        self.ep_conn = ep_conn
+    def __init__(self, ep_conn):
+        super().__init__(
+            action=lambda item: self.ep_conn.send(item),
+            name="StompManager",
+            logger_name=".".join([self.__module__, self.__class__.__name__]),
+        )
+
         self.conn_dict = {}
-        if stomp_config:
-            host_and_ports = [(stomp_config.get("host"), stomp_config.get("port"))]
-            subscribe_destination = stomp_config.get("subscribe_destination")
-            ssl = stomp_config.get("ssl")
-            headers = []
-            if stomp_config.get("headers"):
-                headers = [self.convert_header_to_dict(stomp_config.get("headers"))]
-            self.conn_dict = {
-                f"{host_and_ports}{subscribe_destination}{ssl.get('use_ssl')}": {
-                    "conn": self.connect(
-                        stomp_config, [{"name": name, "main": is_main}]
-                    ),
-                    "gardens": [{"name": name, "main": is_main}],
-                    "headers_list": headers,
-                }
+        self.ep_conn = ep_conn
+
+    def add_connection(self, stomp_config=None, name=None, is_main=False):
+        host_and_ports = [(stomp_config.get("host"), stomp_config.get("port"))]
```

Comment:
Need to add some logic to check if `stomp_config` has `subscribe_destination` and is not `None`. Since the child can be configured in the UI to set `subscribe_destination` to `None`, which means the child needs to send to a topic the entry point is already listening to.

Reply:
I'm still not 100% sure why this is the case. Can you add a commit with your preferred way of fixing this?

Reply:
Sweet, thank you.